controllers/disruption_controller.go

```diff
@@ -272,6 +334,132 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+// createInstanceSelectorCache creates this instance's cache if it doesn't exist and attaches it to the controller
+func (r *DisruptionReconciler) manageInstanceSelectorCache(instance *chaosv1beta1.Disruption) error {
+	// clean up potential expired cache contexts based on context error return
+	deletionList := []string{}
+
+	for key, contextTuple := range r.CacheContextStore {
+		if contextTuple.Ctx.Err() != nil {
+			deletionList = append(deletionList, key)
+		}
+	}
+
+	for _, key := range deletionList {
+		delete(r.CacheContextStore, key)
+	}
+
+	// remove check when StaticTargeting is defaulted to false
+	if instance.Spec.StaticTargeting == nil {
+		r.log.Debugw("StaticTargeting pointer is nil")
```

Comment:
Worth including info about the `instance` here, maybe at least the name?

Reply:
Or are we guaranteed that `r.log` is already set to include those values correctly?

Reply:
The logger [includes the instance name + namespace](https://github.com/DataDog/chaos-controller/blob/1d77e8d736679c6049640692de4f841e1decb21e/controllers/disruption_controller.go#L141) in every log